hummingbot/market/kraken/kraken_api_order_book_data_source.py

```diff
@@ -0,0 +1,299 @@
+            exchange_response: aiohttp.ClientResponse = exchange_response
+            if exchange_response.status != 200:
+                raise IOError(f"Error fetching Kraken exchange information. "
+                            f"HTTP status is {exchange_response.status}.")
+            exchange_data = await exchange_response.json()
+            pairs = []
+            trading_pairs = {}
+            keys = list(exchange_data['result'].keys())
+            for i in range(0, len(keys)):
+                id = keys[i]
+                market = exchange_data['result'][id]
+                baseId = market['base']
+                quoteId = market['quote']
+                base = baseId
+                quote = quoteId
+                symbol = market['altname']
+                trading_pairs[id] = {'symbol': symbol, "baseAsset": base, "quoteAsset": quote, 'altname': market['altname']}
+                pairs.append(market['altname'])
+            params = '&pair='
+            params +=','.join(pairs)
+            exchange_data_ = []
+            exchange_data_.append(exchange_data['result'])
+            market_response = await client.request(url=KRAKEN_TICKER_URL,params=params,method='GET')
+            market_data = await market_response.json()
+            if market_response.status != 200:
+                raise IOError(f"Error fetching Kraken markets information. "
+                            f"HTTP status is {market_response.status}.")
+            market_data: List[Dict[str, Any]] = [
+                {**item[1], **trading_pairs[item[0]]}
+                for item in market_data['result'].items()
+                    if item[0] in trading_pairs
+            ]
+            for colum in market_data:
+                colum['quoteVolume'] = colum['v'][1]
+                colum.pop('v')
+                colum.pop('a')
+                colum.pop('b')
+                colum['close'] = colum['c'][0]
+                colum.pop('c')
+                colum['hcolumgh']= colum['h'][0]
+                colum.pop('h')
+                colum['low'] = colum['l'][0]
+                colum.pop('l')
+                colum['open'] = colum['o'][0]
+                colum.pop('o')
+            all_markets: pd.DataFrame = pd.DataFrame.from_records(data=market_data, index="symbol")
+            btc_price: float = float(all_markets.loc["XBTUSD"]["close"])
+            eth_price: float = float(all_markets.loc["ETHUSD"]["close"])
+            usd_volume: float = [(
+                    quoteVolume * btc_price if symbol.endswith("BTC") else
+                    quoteVolume * eth_price if symbol.endswith("ETH") else
+                    quoteVolume
+                )
+                for symbol, quoteVolume in zip(all_markets.index,
+                                            all_markets.quoteVolume.astype("float"))]
+            all_markets.loc[:, "USDVolume"] = usd_volume
+            all_markets.loc[:, "volume"] = all_markets.quoteVolume
```

Comment:
No need for this

Reply:
What specifically is not necessary? The columns of the dataframe?

Reply:
Line 117